### PR TITLE
Burn down the website by downgrading the JSON schema to draft 07

### LINKED ISSUE #4200: Editor always warns "Draft 2020-12 schemas are not yet fully supported. (768)" in JSON files
Go to [the editor](https://usaco.guide/editor) and open any module.problems.json. The warning will be on the first character.

### PR DESCRIPTION
**Pull request by IDON-TEXIST**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [ ] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [X] I have linked this PR to any issues that it closes.

Closes #4200?

### CODE REVIEW

**Comment by IDON-TEXIST** [comment]
The result: significantly more warnings. Figures.

**Comment by IDON-TEXIST** [comment]
...which doesn't seem to be VScode's fault. It looks like now that we're using a schema that VSCode supports warnings, it's been revealed our JSON is terrible. Looks like an easy enough fix...

**Comment by SansPapyrus683** [comment]
what's wrong w/ our json lol

**Comment by IDON-TEXIST** [comment]
@SansPapyrus683 The "Kind" and "source" fields are enums, but not all the values that we use are in the enum. Why that works at all is beyond me.

File: static/problems.schema.json [hunk 1]
```diff
@@ -34,9 +34,39 @@
...
+            "CEOI",
+            "COI",
+            "COCI",
             "IOI",
-            "___TODO: add more here"
+            "IZhO",
+            "JOI",
+            "LMiO",
+            "RMI",
+            "NOI.sg"
```
Comment (SansPapyrus683) [inline static/problems.schema.json:69]: .sg?

Comment (IDON-TEXIST) [reply static/problems.schema.json:69]: It's what /src/models/problem.ts uses.

**Comment by IDON-TEXIST** [comment]
vercel doesn't seem to recognize the change, and I can't get local building to work.

**Comment by IDON-TEXIST** [comment]
Oh, I get it. The editor is hard-coded to pull files from here, isn't it.
Well that makes things a bit hard to test. opening the JSON files on actual VSCode doesn't warn, so it's probably fine.

**Review by SansPapyrus683 — APPROVED** [review]
seems fine
just give the signal to merge and i'll do it

**Comment by IDON-TEXIST** [comment]
Sure, but be ready to reverse it.

**Review by thecodingwizard — CHANGES_REQUESTED** [review]
why http instead of https?

**Comment by IDON-TEXIST** [comment]
There isn't, I just copied that url from the VSCode docs.

**Review by thecodingwizard — APPROVED** [review]
lgtm, thanks so much for fixing this!! feel free to merge whenever :)

**Pull request merged**